yamux/src/connection.rs

```diff
@@ -490,10 +498,16 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Active<T> {
         }
     }
 
-    fn new_outbound(&mut self) -> Result<Stream> {
+    fn poll_new_outbound(&mut self, cx: &mut Context<'_>) -> Poll<Result<Stream>> {
         if self.streams.len() >= self.config.max_num_streams {
             log::error!("{}: maximum number of streams reached", self.id);
-            return Err(ConnectionError::TooManyStreams);
+            return Poll::Ready(Err(ConnectionError::TooManyStreams));
+        }
```

Comment:
Oh, do I understand correctly that we have been closing the connection when the maximum number of streams have been reached? Does that surprise you as well @thomaseizinger. Wouldn't one expect `poll_new_outbound` to return `Poll::Pending` until one of the existing streams closes and thus one can open a new one?

Reply:
I can time-box and see if I can change this behaviour if you'd like. We are actually not using this variant anywhere else so if we decide to return `Poll::Pending` here, we could get rid of this error variant.

There doesn't seem to be a limit on how many streams the remote can open to us which is rather surprising as well.

Reply:
Thanks for expanding on this. I suggest only tackling this in case we see it in the wild. Also definitely as a follow-up.